### PR TITLE
replace api endpoint

### PR DESCRIPTION
**Pull request by samweinberg23**
- Add replace api endpoint
- Abstracts jsonifying order

### CODE REVIEW

File: src/modules/Api.ts [hunk 1]
```diff
@@ -75,6 +75,84 @@ export class Api {
...
+    makerMarket: Integer | string,
+    takerMarket: Integer | string,
+    makerAmount: Integer | string,
+    takerAmount: Integer | string,
+    expiration: Integer | string,
+    fillOrKill: boolean,
+    cancelId: string,
+    clientId?: string,
+  }): Promise<{ order: ApiOrder }> {
+    const order: SignedLimitOrder = await this.createOrder({
```
Comment (AntonioJuliano) [inline src/modules/Api.ts]: do this and the cancel signing in parallel

File: src/modules/Api.ts [hunk 2]
```diff
@@ -75,6 +75,84 @@ export class Api {
     });
   }
 
+  /**
+   * Creates, but does not place a signed order and signed cancel pair
+   */
+  public async replaceOrder({
```
Comment (AntonioJuliano) [inline src/modules/Api.ts]: Can we break this out into two functions: this one and one that takes an already signed order and cancel and just makes the API request?

File: src/modules/Api.ts [hunk 3]
```diff
@@ -75,6 +75,114 @@ export class Api {
...
+    return this.submitReplaceOrder({
+      order,
+      fillOrKill,
+      cancelId,
+      cancelSignature,
+      clientId,
+    });
+  }
+
+    /**
```
Comment (BrendanChou) [inline src/modules/Api.ts]: nit: extra indent

File: src/modules/Api.ts [hunk 4]
```diff
@@ -75,6 +75,114 @@ export class Api {
...
+      takerAccountOwner: order.takerAccountOwner,
+      takerAccountNumber: order.takerAccountNumber.toFixed(0),
+      makerMarket: order.makerMarket.toFixed(0),
+      takerMarket: order.takerMarket.toFixed(0),
+      makerAmount: order.makerAmount.toFixed(0),
+      takerAmount: order.takerAmount.toFixed(0),
+      salt: order.salt.toFixed(0),
+      expiration: order.expiration.toFixed(0),
+    };
+  }
```
Comment (BrendanChou) [inline src/modules/Api.ts]: nit: place helper functions at the bottom of the file

File: src/modules/Api.ts [hunk 5]
```diff
@@ -75,6 +75,114 @@ export class Api {
...
+
+    const data: any = {
+      order: jsonOrder,
+    };
+    if (clientId) {
+      data.clientId = clientId;
+    }
+    data.fillOrKill = !!fillOrKill;
+    data.cancelId = cancelId;
+    data.cancelSignature = cancelSignature;
```
Comment (BrendanChou) [inline src/modules/Api.ts]: nit: do

```
const data: any = {
  order: jsonOrder,
  cancelId,
  cancelSignature,
  fillOrKill: !!fillOrKill,
};
if (clientId) {
  data.clientId = clientId;
}
```

**Pull request merged**